File: goal-keeper-backend/src/main/java/com/goalkeeper/api/dto/PlayerDataDto.java
package com.goalkeeper.api.dto;

import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Builder
@Data
@JsonRootName("playerData")
@NoArgsConstructor
@AllArgsConstructor
public class PlayerDataDto {

    private UUID id;
    private String fullName;
    private Integer shirtNumber;
    private String position;
    private Integer totalSessions;
    private Integer totalGoals;
    private String winPercentage;
    private String tiePercentage;
    private String lossPercentage;

}

File: goal-keeper/src/data/dto/PlayerDataDto.ts
import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject('PlayerDataDto')
export default class PlayerDataDto {

    @JsonProperty('id', String)
    public id: string | undefined = undefined;

    @JsonProperty('fullName', String)
    public fullName: string | undefined = undefined;

    @JsonProperty('shirtNumber', Number)
    public shirtNumber: number | undefined = undefined;

    @JsonProperty('position', String)
    public position: string | undefined = undefined;

    @JsonProperty('totalSessions', String)
    public totalSessions: string | undefined = undefined;

    @JsonProperty('totalGoals', Number)
    public totalGoals: number | undefined = undefined;

    @JsonProperty('winPercentage', String)
    public winPercentage: string | undefined = undefined;

    @JsonProperty('winPercentage', String)
    public tiePercentage: string | undefined = undefined;

    @JsonProperty('winPercentage', String)
    public lossPercentage: string | undefined = undefined;
}
